runtime/ast/type.go

```diff
@@ -314,6 +454,22 @@ func (t *ReferenceType) EndPosition() Position {
 	return t.Type.EndPosition()
 }
 
+const referenceTypeAuthKeywordSpaceDoc = prettier.Text("auth ")
+const referenceTypeSymbolDoc = prettier.Text("&")
+
+func (t *ReferenceType) Doc() prettier.Doc {
+	var doc prettier.Concat
+	if t.Authorized {
+		doc = append(doc, referenceTypeAuthKeywordSpaceDoc)
+	}
+
+	return append(
```

Comment:
same question as in the prev PR, should `prettier.Concat{}` be used here?

Reply:
`prettier.Concat`, like other documents, can be nested, so this could be `return prettier.Concat{doc, ...` here too, but there is no semantic difference between nesting of flattening them, expect for the latter being more efficient.